### PR TITLE
[PEPC-Boost] Integrate submit ToB txs api with Tob validation RPC

### LINKED ISSUE #21: Simulate ToB txs sent by searchers
Currently, searchers send their ToB txs to `/relay/v1/builder/tob_txs` endpoint in the mev relayer. The relayer stores the highest value ToB txs, where the value of the ToB txs is defined by how much the searchers pay the proposer. 
To make this API more robust, we need to validate the ToB tx before storing It in our database. We can validate the ToB tx simply by an `eth_estimate_gas` rpc call. We don't need a special validation RPC like `validateBlockSubmission`

### PR DESCRIPTION
**Pull request by bharath-123**
## 📝 Summary

We are now using the TOB validation RPC implemented in the builder here: https://github.com/bharath-123/pepc-boost-builder/pull/15  to validate TOB txs submitted by the searchers. This is a more robust TOB validation compared to what we previously had.

In this PR we integrate the TOB validation RPC in the submit TOB txs flow

Resolves https://github.com/bharath-123/pepc-boost-relay/issues/21

### CODE REVIEW

File: services/api/pepc_boost_test.go [hunk 1]
```diff
@@ -583,10 +591,32 @@ func TestNetworkIndependentCheckTxAndSenderValidity(t *testing.T) {
...
+				"Content-Type": "application/json",
+			})
 
-			err := backend.relay.checkTxAndSenderValidity(c.txs, headSlot+1, common.TestLog)
 			if c.requiredError != "" {
-				require.Contains(t, err.Error(), c.requiredError)
+				require.Equal(t, http.StatusBadRequest, rr.Code)
+				require.Contains(t, rr.Body.String(), c.requiredError)
 			} else {
 				require.NoError(t, err)
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:621]: nit: return in the if and then the `require.NoError` can be in the outer scope (no else)

File: services/api/pepc_boost_test.go [hunk 2]
```diff
@@ -536,44 +550,38 @@ func TestNetworkIndependentCheckTxAndSenderValidity(t *testing.T) {
...
 			},
-			callTraces:    nil,
-			requiredError: "we support only 1 tx on the TOB currently, got 3",
+			callTraces:         nil,
+			tobSimulationError: "contract creation txs are not allowed",
+			requiredError:      "contract creation txs are not allowed",
 		},
 		{
-			description: "First tx is a contract creation",
+			description: "payout not addresses to relayer",
```
Comment (michaelneuder) [inline services/api/pepc_boost_test.go:563]: can you remind me why the payout can't go directly to the validator ? i remember for the RoB if we remove builders transactions then we want to decrease the payout, but the ToB payment is always forwarded to the proposer right ?

File: services/api/blocksim_ratelimiter.go [hunk 3]
```diff
@@ -50,6 +51,33 @@ func NewBlockSimulationRateLimiter(blockSimURL string) *BlockSimulationRateLimit
 	}
 }
 
+func (b *BlockSimulationRateLimiter) TobSim(context context.Context, tobValidationRequest *common.TobValidationRequest) (requestErr, validationErr error) {
```
Comment (michaelneuder) [inline services/api/blocksim_ratelimiter.go:54]: nit: this looks a lot like `Send` can we make a more generic send and pass in the path ?

**Review by michaelneuder — COMMENTED** [review]
LGTM!

File: services/api/pepc_boost_test.go (hunk 2 above)
Comment (bharath-123) [reply services/api/pepc_boost_test.go:563]: the payout does go to the proposer!! I worded it wrong! will correct it!

File: services/api/blocksim_ratelimiter.go (hunk 3 above)
Comment (bharath-123) [reply services/api/blocksim_ratelimiter.go:54]: yeah thats a good idea. will refactor the code.

**Comment by bharath-123** [comment]
i will take up code refactoring of the ToB sim part in another PR @michaelneuder I am planning a big refactor of all RPC calls method. I ll take it up all at once.

**Pull request merged**